js/main.js

```diff
@@ -22,11 +22,87 @@ const getRandomFractionalNumber = function (min, max, symbolsAfterPoint) {
 
   let randomNumber = (getRandomNumber(min, max)).toFixed(symbolsAfterPoint);
   // If the number is fractional and ends in 0
-  if(randomNumber.slice(-1) === '0' && ((randomNumber ^ 0) === randomNumber) === false) {
+  if(randomNumber.slice(-1) === '0' && randomNumber % 1 !== 0) {
     randomNumber = randomNumber.substring(0, randomNumber.length - 1) + getRandomWholeNumber(1, 9) + randomNumber.substring(1 + randomNumber.length - 1);
   }
-  return Number(randomNumber);
+  return +randomNumber;
 }
 
-getRandomWholeNumber(0, 1);
-getRandomFractionalNumber(0, 2, 1);
+const TYPES = [
+  'palace',
+  'flat',
+  'house',
+  'bungalow',
+];
+
+const TIMES = [
+  '12:00',
+  '13:00',
+  '14:00',
+];
+
+const FEATURES = [
+  'wifi',
+  'dishwasher',
+  'parking',
+  'washer',
+  'elevator',
+  'conditioner',
+];
+
+const DESCRIPTIONS = [
+  'What is Lorem Ipsum?',
+  'Why do we use it?',
+  'Where can I get some?',
+  'Where does it come from?',
+];
+
+const IMAGES = [
+  'http://o0.github.io/assets/images/tokyo/hotel1.jpg',
+  'http://o0.github.io/assets/images/tokyo/hotel2.jpg',
+  'http://o0.github.io/assets/images/tokyo/hotel3.jpg',
+];
+
+const featuresUsed = [];
+for (let i = 0; i <= getRandomWholeNumber(0, FEATURES.length); i++) {
+  featuresUsed[i] = FEATURES[i];
+}
+
+const photosUsed = [];
+for (let i = 0; i <= getRandomWholeNumber(0, IMAGES.length); i++) {
+  photosUsed[i] = IMAGES[i];
+}
+
+const createDataKeksobooking = () => {
+  let author = {
+    avatar: `img/avatars/user0${getRandomWholeNumber(1, 8)}.png`,
+  };
+  let offer = {
+    title: DESCRIPTIONS[getRandomWholeNumber(0, DESCRIPTIONS.length - 1)],
+    address: `${getRandomFractionalNumber(35.65, 35.7, 5)}, ${getRandomFractionalNumber(139.7, 139.8, 5)}`,
+    price: getRandomWholeNumber(1000, 10000),
+    type: TYPES[getRandomWholeNumber(0, TYPES.length - 1)],
+    rooms: getRandomWholeNumber(1, 10),
+    guests: getRandomWholeNumber(1, 30),
+    checkin: TIMES[getRandomWholeNumber(0, TIMES.length - 1)],
+    checkout: TIMES[getRandomWholeNumber(0, TIMES.length - 1)],
+    features: featuresUsed,
+    description: DESCRIPTIONS[getRandomWholeNumber(0, DESCRIPTIONS.length - 1)],
+    photos: photosUsed,
+  };
+
+  let location = {
+    x: getRandomFractionalNumber(35.65, 35.7, 5),
+    y: getRandomFractionalNumber(139.7, 139.8, 5),
+  };
+
+  return {
+    author,
+    offer,
+    location,
+  }
```

Comment:
Создание переменных должно быть оправданно, т.к. новая переменная занимает место в памяти, пусть даже временно. Вдобавок разработчик будет искать причину зачем для простых структурных блоков заводится переменная и не найдет ее.
Тут можно обойтись без них:
```suggestion
  return {
    author: {
      avatar: `img/avatars/user0${getRandomWholeNumber(1, 8)}.png`,
    },
    offer: {...},
    location: {...}
  }
```